x/nameservice/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,269 @@
+package keeper_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/tharsis/ethermint/x/nameservice/client/cli"
+	nameservicetypes "github.com/tharsis/ethermint/x/nameservice/types"
+	"os"
+)
+
+func (suite *KeeperTestSuite) TestGrpcQueryParams() {
+	grpcClient := suite.queryClient
+
+	testCases := []struct {
+		msg string
+		req *nameservicetypes.QueryParamsRequest
+	}{
+		{
+			"Get Params",
+			&nameservicetypes.QueryParamsRequest{},
+		},
+	}
+	for _, test := range testCases {
+		suite.Run(fmt.Sprintf("Case %s ", test.msg), func() {
+			resp, _ := grpcClient.Params(context.Background(), test.req)
+			defaultParams := nameservicetypes.DefaultParams()
+			suite.Require().Equal(defaultParams.String(), resp.GetParams().String())
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestGrpcGetRecordLists() {
+	grpcClient, ctx := suite.queryClient, suite.ctx
+	sr := suite.Require()
+	var recordId string
+	testCases := []struct {
+		msg          string
+		req          *nameservicetypes.QueryListRecordsRequest
+		createRecord bool
+		expErr       bool
+		noOfRecords  int
+	}{
+		{
+			"Empty Records",
+			&nameservicetypes.QueryListRecordsRequest{},
+			false,
+			false,
+			0,
+		},
+		{
+			"List Records",
+			&nameservicetypes.QueryListRecordsRequest{},
+			true,
+			false,
+			1,
+		},
+	}
+	for _, test := range testCases {
+		suite.Run(fmt.Sprintf("Case %s ", test.msg), func() {
+			if test.createRecord {
+				dir, err := os.Getwd()
+				sr.NoError(err)
+				payload, err := cli.GetPayloadFromFile(dir + "/../helpers/examples/example1.yml")
+				sr.NoError(err)
+				err = suite.app.NameServiceKeeper.ProcessSetRecord(ctx, nameservicetypes.MsgSetRecord{
+					BondId:  suite.bond.GetId(),
+					Signer:  suite.accounts[0].String(),
+					Payload: payload.ToPayload(),
+				})
+				sr.NoError(err)
+			}
+			resp, err := grpcClient.ListRecords(context.Background(), test.req)
+			if test.expErr {
+				suite.Error(err)
+			} else {
+				sr.NoError(err)
+				sr.Equal(test.noOfRecords, len(resp.GetRecords()))
+				if test.createRecord {
+					recordId = resp.GetRecords()[0].GetId()
+					sr.NotZero(resp.GetRecords())
+					sr.Equal(resp.GetRecords()[0].GetBondId(), suite.bond.GetId())
+				}
+			}
+		})
+	}
+
+	// Get the records by record id
```

Comment:
Maybe add a log for these different record fetching scenarios, similar to what you did in `client/testutil/query.go`.